components/flex/nz-flex.directive.ts

```diff
@@ -0,0 +1,57 @@
+/**
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://github.com/NG-ZORRO/ng-zorro-antd/blob/master/LICENSE
+ */
+
+import { Directive, Input } from '@angular/core';
+
+import { NzAlign, NzFlex, NzGap, NzJustify, NzWrap } from './typings';
+
+@Directive({
+  selector: '[nz-flex],nz-flex',
+  exportAs: 'nzFlex',
+  standalone: true,
+  host: {
+    class: 'ant-flex',
+    '[class.ant-flex-vertical]': `nzVertical`,
+    '[class.ant-flex-justify-flex-start]': `nzJustify === 'flex-start'`,
+    '[class.ant-flex-justify-center]': `nzJustify === 'center'`,
+    '[class.ant-flex-justify-flex-end]': `nzJustify === 'flex-end'`,
+    '[class.ant-flex-justify-space-between]': `nzJustify === 'space-between'`,
+    '[class.ant-flex-justify-space-around]': `nzJustify === 'space-around'`,
+    '[class.ant-flex-justify-space-evenly]': `nzJustify === 'space-evenly'`,
+    '[class.ant-flex-justify-start]': `nzJustify === 'start'`,
+    '[class.ant-flex-justify-end]': `nzJustify === 'end'`,
+    '[class.ant-flex-justify-right]': `nzJustify === 'right'`,
+    '[class.ant-flex-justify-left]': `nzJustify === 'left'`,
+    '[class.ant-flex-justify-stretch]': `nzJustify === 'stretch'`,
+    '[class.ant-flex-justify-normal]': `nzJustify === 'normal'`,
+    '[class.ant-flex-align-flex-start]': `nzAlign === 'flex-start'`,
+    '[class.ant-flex-align-center]': `nzAlign === 'center'`,
+    '[class.ant-flex-align-flex-end]': `nzAlign === 'flex-end'`,
+    '[class.ant-flex-align-space-between]': `nzAlign === 'space-between'`,
+    '[class.ant-flex-align-space-around]': `nzAlign === 'space-around'`,
+    '[class.ant-flex-align-space-evenly]': `nzAlign === 'space-evenly'`,
+    '[class.ant-flex-align-start]': `nzAlign === 'start'`,
+    '[class.ant-flex-align-end]': `nzAlign === 'end'`,
+    '[class.ant-flex-align-right]': `nzAlign === 'right'`,
+    '[class.ant-flex-align-left]': `nzAlign === 'left'`,
+    '[class.ant-flex-align-stretch]': `nzAlign === 'stretch'`,
+    '[class.ant-flex-align-normal]': `nzAlign === 'normal'`,
+    '[class.ant-flex-wrap-wrap]': `nzWrap === 'wrap'`,
+    '[class.ant-flex-wrap-wrap-reverse]': `nzWrap === 'wrap-reverse'`,
+    '[class.ant-flex-wrap-nowrap]': `nzWrap === 'nowrap'`,
+    '[style.gap]': `nzGap === 'small' ? '8px' : nzGap === 'middle' ? '16px' : nzGap === 'large' ? '24px' : (Number.isInteger(nzGap) ? nzGap + 'px' : nzGap)`,
```

Comment:
You cannot use `Number.isInteger()` directly in a template expression. I think we should use a `getter` at this point:

```ts
`[style.gap]`: `gap`

protected get gap() {
  switch (this.nzGap) {
    case 'small': return '8px';
    ...
    default:
      if (typeof this.nzGap === 'number') {
        return this.nzGap + 'px';
      }
      return this.nzGap;
  }
}
```

Reply:
👌

Reply:
This approach has a drawback, as it invokes a method directly in the template expression. This means that the method is called every time a change detection cycle occurs.

see this article(6.1 Template Methods): <a href="https://blog.oasisdigital.com/2017/angular-runtime-performance-guide/">Angular Runtime Performance Guide</a>

Reply:
I check if an implementation of pipe can prove to be useful.

Reply:
Don't worry about performance, since this getter requires very little computation and it's fast.

Reply:
Or you can handle it in ngOnChanges. Because I don't want to add a new pipe here.

Reply:
I will use the getter method then